### PR TITLE
Ensure numbers are in safe range

### PR DESCRIPTION
**Pull request by greim**
This is an attempt to fix this: https://github.com/Netflix/falcor-router/issues/176

### CODE REVIEW

**Comment by ThePrimeagen** [comment]
@greim this seems to be correct.  For funsies could you make sure you put a unit test with a string number that is too big.  I want to make sure that the behavior of what we are fixing is in a unit test.  If you do that, i'll do the merging.

File: test/toPaths.spec.js [hunk 1]
```diff
@@ -149,5 +151,65 @@ describe('toPaths', function() {
...
+            true,
+            false,
+            undefined,
+            Infinity,
+            -Infinity,
+            '9007199254740992', // max safe int + 1
+            '-9007199254740992', // min safe int - 1
+            9007199254740992, // max safe int + 1
+            -9007199254740992, // min safe int - 1
+            '648365838265483646384563538',
```
Comment (greim) [inline test/toPaths.spec.js:235]: @michaelbpaulson - Yup, it has some numbers that are out of the safe range by 1, and now also an absurdly large string number.

**Comment by ThePrimeagen** [comment]
Could you do a collapse just to have a test that describe the behavior we want and ensure that its working.

**Comment by greim** [comment]
@michaelbpaulson Not sure I follow. You mean add a test that calls `toPaths()` on something that will invoke the specific behavior we want? (I.e. in addition to testing `isSafeNumber()` directly)

File: test/toPaths.spec.js [hunk 2]
```diff
@@ -68,6 +70,46 @@ describe('toPaths', function() {
...
+        expect(paths[1][1][0]).to.equal(0);
+        expect(paths[1][1][1]).to.equal(1);
+        expect(paths[1][1][2]).to.equal(234);
+        expect(paths[1][1][3]).to.equal(345678);
+        expect(paths[1][1][4]).to.equal(4253674286);
+        expect(paths[1][1][5]).to.equal(9007199254740991); // max safe int
+        expect(paths[1][1][6]).to.equal('9007199254740992'); // max safe int + 1
+        expect(paths[1][1][7]).to.equal('918572487653498743278645'); // absurdly large
+    });
+
```
Comment (greim) [inline test/toPaths.spec.js:112]: @michaelbpaulson Hopefully that captures it. Let me know if not.

**Comment by jameslaneconkling** [comment]
@michaelbpaulson is there any movement on this?  

Ran into this issue when requesting a resource with an id that is a stringified large int.  E.g.

``` javascript
falcor.get(["users", "994717620572244942", "label"])
```

sends out a request as

```
paths:[["users",994717620572245000,"label"]]
method:get
```

What gets stored in the falcor cache is the correct key, but what comes back is of course the incorrect key, so falcor issues a `MaxRetryExceededError`.

I'll apply this PR to my local version to see if it fixes it.

**Comment by jameslaneconkling** [comment]
Can confirm this fixed my issue.  Thanks @greim.

**Comment by lrowe** [comment]
Thanks @greim! I've rebased, added checks for floats and merged.

**Pull request closed without merging**